client/client.go

```diff
@@ -243,80 +214,200 @@ func (c *Client) update(latestRoot bool) (data.TargetFiles, error) {
 		}
 	}
 
-	// Save the snapshot.json now it has been processed successfully
-	if err := c.local.SetMeta("snapshot.json", snapshotJSON); err != nil {
-		return nil, err
-	}
-
 	return updatedTargets, nil
 }
 
-func (c *Client) updateWithLatestRoot(m *data.SnapshotFileMeta) (data.TargetFiles, error) {
-	var rootJSON json.RawMessage
-	var err error
-	if m == nil {
-		rootJSON, err = c.downloadMetaUnsafe("root.json", defaultRootDownloadLimit)
-	} else {
-		rootJSON, err = c.downloadMetaFromSnapshot("root.json", *m)
+func (c *Client) updateRoots() error {
+	// https://theupdateframework.github.io/specification/v1.0.19/index.html#load-trusted-root
+	// Temporarily deactivate expiration check. This is consistent with 5.3.6:
+	// Note that the expiration of the trusted root metadata file does not
+	// matter, because we will attempt to update it in the next step.
+	e := verify.IsExpired
+	defer func() { verify.IsExpired = e }()
+	verify.IsExpired = func(t time.Time) bool { return false }
+
+	// 5.2 Load the trusted root metadata file. We assume that a good,
+	// trusted copy of this file was shipped with the package manager
+	// or software updater using an out-of-band process.
+	if err := c.loadAndVerifyLocalRootMeta(); err != nil {
+		return err
 	}
+	m, err := c.local.GetMeta()
 	if err != nil {
-		return nil, err
+		return err
 	}
-	if err := c.decodeRoot(rootJSON); err != nil {
-		return nil, err
+
+	type KeyInfo struct {
+		KeyIDs    map[string]bool
+		Threshold int
 	}
-	if err := c.local.SetMeta("root.json", rootJSON); err != nil {
-		return nil, err
+
+	// Prepare for 5.3.11: If the timestamp and / or snapshot keys have been rotated,
+	// then delete the trusted timestamp and snapshot metadata files.
+	getKeyInfo := func(role string) KeyInfo {
+		//keyIDs := make([]string, 0, len(c.db.GetRole(role).KeyIDs))
```

Comment:
nit: remove these commented out lines of code from old versions

Reply:
Done!